tony-history-server/app/utils/HdfsUtils.java

```diff
@@ -0,0 +1,108 @@
+package utils;
+
+import java.io.BufferedReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import play.Logger;
+
+
+/**
+ * The class handles all HDFS file operations
+ */
+public class HdfsUtils {
+  private static final Logger.ALogger LOG = Logger.of(HdfsUtils.class);
+
+  static boolean isPathValid(FileSystem fs, Path filePath) {
+    try {
+      return fs.exists(filePath);
+    } catch (IOException e) {
+      LOG.error(filePath.toString() + " doesn't exist!");
+      return false;
+    }
+  }
+
+  static String contentOfHdfsFile(FileSystem fs, Path filePath) {
+    StringBuilder fileContent = new StringBuilder();
+    try (FSDataInputStream inStrm = fs.open(filePath);
+        BufferedReader bufReader = new BufferedReader(new InputStreamReader(inStrm))) {
+      String line;
+      while ((line = bufReader.readLine()) != null) {
+        fileContent.append(line);
+      }
+      return fileContent.toString();
+    } catch (IOException e) {
+      LOG.error("Couldn't read content of file from HDFS", e);
+      return "";
+    }
+  }
+
+  static String copyFromHdfs(FileSystem fs, Path hdfsFilePath) {
+    String localFileName = "config.xml";
+    try (FSDataInputStream inStrm = fs.open(hdfsFilePath); FileWriter fw = new FileWriter(localFileName)) {
+      IOUtils.copy(new InputStreamReader(inStrm), fw);
+    } catch (IOException e) {
+      LOG.error("Couldn't copy from file from HDFS", e);
+      return "";
+    }
+    return localFileName;
+  }
+
+  static List<Path> getValidPaths(FileStatus[] lsJobDir, Predicate<FileStatus> fn) {
```

Comment:
We need tests for methods in this class

Reply:
I'll look into how tests are written for a Play application and I'll add some soon.

Reply:
These utility methods should be easy to test since you can just instantiate the input objects, directly call the methods, and then assert things about the returned output.